contracts/Donations.sol

```diff
@@ -91,16 +91,13 @@ contract Donations is ERC721, AccessControl {
      * @param _to Address of the charity.
      */
     function donate(
-        uint256 _destinationId,
+        uint128 _destinationId,
```

Comment:
Why this change?

Reply:
If you check the Metadata struct, then you will find that i have change the destinationId type there to uint128 from uint256. And I also moved the "uint64 expiry" struct to 2nd position from 4th position. What this will do this pack the destinationId & the expiry variables in the same 32 bytes slot instead of using 2 slots for them.  Due to this change I also had to change the transferableAmounts mapping to use uint128 for the destinationId. Thus, I changed it to uint128 in the donate method too. Else if it was uint256 here, then we would have to cast it to uint128 again to use in the transferableAmounts  update on line 101. ( I am not totally sure about the last part, maybe if we kept it at uint256 then solidity would internally cast it to uint128 on line 101, but not sure if that would use more gas or not)

Reply:
sorry, I wasn't asking about this line specifically, but I understand that you did it to be able to pack it with the other variable. My concern is that I'm not sure 128 is enough, [their API](https://the-giving-block.gitbook.io/public-api-documentation/endpoints-documentation) doesn't specify the size of the ID

Reply:
let's ask them if there's a cap :]

Reply:
I guess they wont make ids larger than 340282366920938463463374607431768211456 (2**128). But not sure, definitely a good idea to be double sure by asking them.

Reply:
Alex from TGB said he doesn't think so (and if he comes back and rectifies it we can fix it anyway)

Reply:
so what do we do ? uint256, uint128 or we can also try uint192? Since uint192 will still total up to 256 when packed with uint64.

Reply:
either one but it must add up to 256 or it will use more gas

Reply:
You can keep it as is. It's been confirmed 👍